src/core/blueprint/index.rs

```diff
@@ -66,6 +66,37 @@ impl Index {
     pub fn get_mutation(&self) -> Option<&str> {
         self.schema.mutation.as_deref()
     }
+
+    pub fn is_type_implements(&self, type_name: &str, implements: &str) -> bool {
+        if type_name == implements {
+            return true;
+        }
+
+        if !matches!(
+            self.map.get(implements),
+            Some((Definition::Interface(_), _))
+        ) {
+            return false;
+        }
+
+        self.is_type_implements_rec(type_name, implements)
```

Comment:
I also feel since this is an index, we should pre-populate the implementations of each type so that finding that a type implements an interface is an O(1) operation and not an `O(N)` which is what it is right now.

Reply:
recursion is removed the check is O(1) now